packages/vega-util/src/flush.ts

```diff
@@ -0,0 +1,37 @@
+import peek from './peek.js';
+
+/**
+ * Selects one of three values (left, right, or center) based on whether
+ * a value is within a threshold distance of a range's endpoints.
+ * @template T The type of value to return (can be any type - string, number, object, etc.)
+ */
+export default function<T>(
+  range: readonly number[],
+  value: number,
+  threshold: number | null | undefined,
+  left: T,
+  right: T,
+  center: T
+): T {
+  if (!threshold && threshold !== 0) return center;
+
+  const t = +threshold;
+  let a = range[0],
+      b = peek(range),
+      l: number;
+
+  // TODO: Check in PRV: behavior change to handle empty b
```

Comment:
This is a behavior change from before. It shouldn't happen if range contains at least 1 member, but it will happen if range is an empty array.